content/client-lib-development-guide/features.textile

```diff
@@ -1464,6 +1486,82 @@ enum StatsIntervalGranularity:
   DAY
   MONTH
 
+class DeviceDetails:
+  + fromLocalDevice() => io DeviceDetails
+  id: String
+  platform: DevicePlatform
+  formFactor: DeviceFormFactor
+  clientId: String?
+  metadata: Dict<String, String>
+  updateTokenId: String
+  resetId(String)
+  resetUpdateToken() => io
+  push: DevicePushDetails
+
+class DevicePushDetails:
+  transportType: DevicePushTransportType
+  transportId: String
+  transportToken: String?
+  state: .Active | .Failing | .Failed
+  errorReason: ErrorInfo?
+  metadata: Dict<String, String>
+
+class Push:
+  publish(options: Dict<String, String>, JsonObject) => io
+  admin: PushAdmin
+
+  // Only on platforms that support receiving notifications:
+
+  activate(
+    DeviceDetails,
+    registerCallback: ((ErrorInfo?, DeviceDetails?) -> io String)?
+  ) => io ErrorInfo? // RSH2c
+  deactivate(
+    deviceId: String,
+    deregisterCallback: ((ErrorInfo?, deviceId: String?) -> io)?
+  ) => io ErrorInfo?
+
+class PushAdmin:
+  deviceRegistrations: PushDeviceRegistrations
+  channelSubscriptions: PushChannelSubscriptions
+
+class JsonObject:
+  // Platform-dependant, typically a Dict-like object
+
+class PushDeviceRegistrations:
+  save(DeviceDetails) => io
+  get(params: Dict<String, String>?) => io PaginatedResult<DeviceDetails>
+  remove(params: Dict<String, String>?) => io
+
+class PushChannelSubscriptions:
+  save([PushChannelSubscription]) => io
+  save(PushChannelSubscription) => io
+  listChannels() => io PaginatedResult<String>
+  get(params: Dict<String, String>?) => io PaginatedResult<PushChannelSubscription>
+  remove(params: Dict<String, String>?) => io
+
+enum DevicePushTransportType:
+  "fcm" // PTT1
+  "gcm" // PTT1
+  "apns" // PTT1
+  "web" // PTT1
+
+enum DevicePlatform:
+  "android" // PPT1
+  "ios" // PPT1
+  "browser" // PPT1
+
+enum DeviceFormFactor:
+  "mobile" // PDT1
+  "tablet" // PDT1
+  "desktop" // PDT1
+  "embedded" // PDT1
```

Comment:
BTW I think this isn't really useful as-is, because `embedded` tells you nothing about what the device is, and it just means "none of the above". I suggest we either:

- define something more meaningful here (not sure what that would be - but could for example include `embedded-headless`);

- be honest have have an `other` value;

- remove this and do nothing until we have actual examples of a device type and use-cases that allow us to define something meaningful.